contracts/src/instances/erc4626/ERC4626Hyperdrive.sol

```diff
@@ -67,27 +64,19 @@ contract ERC4626Hyperdrive is Hyperdrive, ERC4626Base {
     /// @param _target2 The target2 address.
     /// @param _target3 The target3 address.
     /// @param _target4 The target4 address.
-    /// @param __vault The ERC4626 compatible yield source.
     constructor(
         IHyperdrive.PoolConfig memory _config,
         address _target0,
         address _target1,
         address _target2,
         address _target3,
-        address _target4,
-        IERC4626 __vault
-    )
-        Hyperdrive(_config, _target0, _target1, _target2, _target3, _target4)
-        ERC4626Base(__vault)
-    {
-        // Ensure that the base token is the same as the vault's underlying
-        // asset.
-        if (address(_config.baseToken) != IERC4626(_vault).asset()) {
-            revert IHyperdrive.InvalidBaseToken();
-        }
-
+        address _target4
+    ) Hyperdrive(_config, _target0, _target1, _target2, _target3, _target4) {
         // Approve the base token with 1 wei. This ensures that all of the
         // subsequent approvals will be writing to a dirty storage slot.
-        ERC20(address(_config.baseToken)).forceApprove(address(_vault), 1);
+        ERC20(address(_config.baseToken)).forceApprove(
```

Comment:
question: why do this only for ERC4626 yield sources?

Reply:
The `ERC4626Hyperdrive` instance is the only integration that supports a large number of tokens. All of the other integrations are 1:1 with the underlying token (for example, `StETHHyperdrive` is always `ETH` and `Lido`). Furthermore, all of the other integrations use `ETH` as their base token, so they don't even need to deal with approvals since they can directly forward ETH.

With this in mind, we also don't need to use `forceApprove` and `safeTransferFrom` since we know that the vaults are ERC20 compliant (we only need these libraries for implementations that aren't ERC20 compliant). With this said, I think we should use `safeTransferFrom` and `forceApprove` as a general policy to cover our ass and add some layers of security.